### PR TITLE
Introduce extensive string length limitations for updateMe

### LINKED ISSUE #31: User Input String Limits
Generally all string should be limited. Ideally to a explicit limit.

Currently missing in `updateMe` for fields:
- `biography`, 2k?
- `pgpPublicKey`, 10k? My 4k public key is 3114 chars
- `publicUrl`, 2048
- `twitterHandle`, 15
- `facebookId`, 64 (can also be something like `profile.php?id=xxxxxxxxxxxxxxx`)
- `phoneNumberNote`, 140
- `phoneNumber`, 20 (15 digits but let's give 5 spaces for formatting, e.g. `0049 XXX XX XX XX XX`)
- `firstName`, 32?
- `lastName`, 32?

### PR DESCRIPTION
**Pull request by pozylon**
closes #31

### CODE REVIEW

File: lib/ensureStringLength.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    max = DEFAULT_MAX,
+    min = DEFAULT_MIN,
+    error = DEFAULT_ERROR
+  } = options
+
+  if (Number.isInteger(max) && length > max) {
+    throw new Error(error)
+  } else if (Number.isInteger(min) && length < min) {
+    throw new Error(error)
+  }
```
Comment (patte) [inline lib/ensureStringLength.js]: why not `parseInt(max)`?

**Review by patte — APPROVED** [review]

File: lib/ensureStringLength.js (hunk 1 above)
Comment (pozylon) [reply lib/ensureStringLength.js]: parseInt is doing too much magic for my taste.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseInt (see examples).

Comment (patte) [reply lib/ensureStringLength.js]: Thats the real beauty of javascript. Meh. I just think it's more robust to parse an int somehow than to just ignore the limits if they are not-exactly-an-int. What about the `filterInt` function from the link you posted?

Comment (pozylon) [reply lib/ensureStringLength.js]: Aight, I take that one

**Comment by patte** [comment]
fyi: I implemented error messages for all fields: a3b0b26ff0b8c200b44d3495caaca4b44801f976

**Pull request merged**